undertow/src/main/java/org/wildfly/extension/undertow/UndertowTransformers.java

```diff
@@ -74,6 +76,14 @@ private static void registerTransformers_EAP_7_1_0(SubsystemTransformerRegistrat
                 .end();
 
         TransformationDescription.Tools.register(subsystemBuilder.build(), subsystemRegistration, MODEL_VERSION_EAP7_1_0);
+
+        final ResourceTransformationDescriptionBuilder serverBuilder = subsystemBuilder.addChildResource(UndertowExtension.SERVER_PATH);
+        final AttributeTransformationDescriptionBuilder http = serverBuilder.addChildResource(UndertowExtension.HTTP_LISTENER_PATH).getAttributeBuilder();
+        convertCommonListenerAttributes(http).end();
+        final AttributeTransformationDescriptionBuilder https = serverBuilder.addChildResource(UndertowExtension.HTTPS_LISTENER_PATH).getAttributeBuilder();
+        convertCommonListenerAttributes(https).end();
+        final AttributeTransformationDescriptionBuilder ajp = serverBuilder.addChildResource(UndertowExtension.AJP_LISTENER_PATH).getAttributeBuilder();
+        convertCommonListenerAttributes(ajp).end();
```

Comment:
Please move this code above L78. It should be create the builder, configure the builder, register the builder.

Reply:
updated.

Reply:
I have updated to add a test to cover it. However, it needs to wait for next WFCORE update to have new ModelTestControllerVersion EAP_7_1_0 made in https://issues.jboss.org/browse/WFCORE-3469

I tested with a SNAPSHOT version, but I got:
```
testConvertTransformersEAP_7_1_0(org.wildfly.extension.undertow.UndertowTransformersTestCase)  Time elapsed: 1.929 sec  <<< ERROR!
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.as.subsystem.bridge.local.ScopedKernelServicesBootstrap.createChildClassLoaderKernelServices(ScopedKernelServicesBootstrap.java:85)
	at org.jboss.as.subsystem.bridge.local.ScopedKernelServicesBootstrap.createKernelServices(ScopedKernelServicesBootstrap.java:53)
	at org.jboss.as.subsystem.test.SubsystemTestDelegate$LegacyKernelServiceInitializerImpl.install(SubsystemTestDelegate.java:771)
	at org.jboss.as.subsystem.test.SubsystemTestDelegate$LegacyKernelServiceInitializerImpl.access$700(SubsystemTestDelegate.java:641)
	at org.jboss.as.subsystem.test.SubsystemTestDelegate$KernelServicesBuilderImpl.build(SubsystemTestDelegate.java:596)
	at org.wildfly.extension.undertow.UndertowTransformersTestCase.doConvertTest(UndertowTransformersTestCase.java:145)
	at org.wildfly.extension.undertow.UndertowTransformersTestCase.testConvertTransformersEAP_7_1_0(UndertowTransformersTestCase.java:80)
Caused by: java.lang.Exception: java.lang.LinkageError: loader constraint violation: when resolving method "org.jboss.as.clustering.controller.ResourceDescriptor.<init>(Lorg/jboss/as/controller/descriptions/ResourceDescriptionResolver;)V" the class loader (instance of org/jboss/as/model/test/ChildFirstClassLoader) of the current class, org/wildfly/extension/undertow/HostSingleSignOnDefinition, and the class loader (instance of sun/misc/Launcher$AppClassLoader) for the method's defining class, org/jboss/as/clustering/controller/ResourceDescriptor, have different Class objects for the type org/jboss/as/controller/descriptions/ResourceDescriptionResolver used in the signature
	at org.jboss.as.model.test.ModelTestModelControllerService.boot(ModelTestModelControllerService.java:288)
	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:417)
	at org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:370)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "org.jboss.as.clustering.controller.ResourceDescriptor.<init>(Lorg/jboss/as/controller/descriptions/ResourceDescriptionResolver;)V" the class loader (instance of org/jboss/as/model/test/ChildFirstClassLoader) of the current class, org/wildfly/extension/undertow/HostSingleSignOnDefinition, and the class loader (instance of sun/misc/Launcher$AppClassLoader) for the method's defining class, org/jboss/as/clustering/controller/ResourceDescriptor, have different Class objects for the type org/jboss/as/controller/descriptions/ResourceDescriptionResolver used in the signature
	at org.wildfly.extension.undertow.HostSingleSignOnDefinition.registerOperations(HostSingleSignOnDefinition.java:45)
	at org.jboss.as.controller.registry.NodeSubregistry.registerChild(NodeSubregistry.java:109)
	at org.jboss.as.controller.registry.ConcreteResourceRegistration.registerSubModel(ConcreteResourceRegistration.java:231)
	at org.jboss.as.controller.PersistentResourceDefinition.registerChildren(PersistentResourceDefinition.java:68)
	at org.jboss.as.controller.registry.NodeSubregistry.registerChild(NodeSubregistry.java:111)
	at org.jboss.as.controller.registry.ConcreteResourceRegistration.registerSubModel(ConcreteResourceRegistration.java:231)
	at org.jboss.as.controller.PersistentResourceDefinition.registerChildren(PersistentResourceDefinition.java:68)
	at org.jboss.as.controller.registry.NodeSubregistry.registerChild(NodeSubregistry.java:111)
	at org.jboss.as.controller.registry.ConcreteResourceRegistration.registerSubModel(ConcreteResourceRegistration.java:231)
	at org.jboss.as.controller.PersistentResourceDefinition.registerChildren(PersistentResourceDefinition.java:68)
	at org.jboss.as.controller.registry.NodeSubregistry.registerChild(NodeSubregistry.java:111)
	at org.jboss.as.controller.registry.ConcreteResourceRegistration.registerSubModel(ConcreteResourceRegistration.java:231)
	at org.jboss.as.controller.extension.ExtensionRegistry$SubsystemRegistrationImpl.registerSubsystemModel(ExtensionRegistry.java:706)
	at org.wildfly.extension.undertow.UndertowExtension.initialize(UndertowExtension.java:105)
	at org.wildfly.legacy.test.controller.subsystem_11_0_0.TestModelControllerService11_0_0.preBoot(TestModelControllerService11_0_0.java:111)
	at org.jboss.as.model.test.ModelTestModelControllerService.boot(ModelTestModelControllerService.java:264)
	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:417)
	at org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:370)
	at java.lang.Thread.run(Thread.java:748)
```

says there are different class objects for ResourceDescriptionResolver are loaded in two classloaders, but I did not make any change to HostSingleSignOnDefinition. Can this happen because of new changes made in latest 4.0.0.Alpha6-SNAPSHOT I used ?

Reply:
I'm not sure why that happened. We should upgrade core tomorrow at which point it will be easier to look at things.

Reply:
@soul2zimate I'm a bit out of the loop on this one, but I don't see this one being moved? This is just based on the original code/comment still showing up and not as part of 'comment on outdated change'

Reply:
I moved the builder configuration before builder registration. The transformer test passed required dependencies.
Can I get a new review? @wildfly/mergers-full